File: tests/demo_trace.py
import asyncio
import time
from zena_mode.arbitrage import SwarmArbitrator


async def demo_trace():
    """Demo trace."""
    # [X-Ray auto-fix] print("[DEBUG] Initializing Arbitrator with Auto-Discovery...")
    arb = SwarmArbitrator()

    if len(arb.endpoints) < 2:
        print("[!] Swarm currently Standalone. Forcing mock expansion for demonstration...")
        # If the user's experts aren't running, we'll mock them pointing to the same endpoint
        # to prove the PARALLEL Logic Trace works.
        base = arb.endpoints[0] if arb.endpoints else "http://127.0.0.1:8001/v1/chat/completions"
        arb.endpoints = [base] * 5
        arb.ports = [8001] * 5

    # [X-Ray auto-fix] print(f"[DEBUG] Active Experts: {len(arb.endpoints)}")
    query = "Why do we use Chain of Thought in LLMs?"
    sys_prompt = "You are an AI Architecture Expert. Be concise."

    print("\n--- STARTING LIVE MULTI-EXPERT TRACE ---\n")
    async for _ in arb.get_cot_response(query, sys_prompt, verbose=False):
        pass


if __name__ == "__main__":
    asyncio.run(demo_trace())

File: rust_output/src/demo_trace.rs
use anyhow::{Result, Context};
use crate::arbitrage::{SwarmArbitrator};
use tokio;

/// Demo trace.
pub async fn demo_trace() -> () {
    // Demo trace.
    println!("{}", "[DEBUG] Initializing Arbitrator with Auto-Discovery...".to_string());
    let mut arb = SwarmArbitrator();
    if arb.endpoints.len() < 2 {
        println!("{}", "[!] Swarm currently Standalone. Forcing mock expansion for demonstration...".to_string());
        let mut base = if arb.endpoints { arb.endpoints[0] } else { "http://127.0.0.1:8001/v1/chat/completions".to_string() };
        arb.endpoints = (vec![base] * 5);
        arb.ports = (vec![8001] * 5);
    }
    println!("[DEBUG] Active Experts: {}", arb.endpoints.len());
    let mut query = "Why do we use Chain of Thought in LLMs?".to_string();
    let mut sys_prompt = "You are an AI Architecture Expert. Be concise.".to_string();
    println!("{}", "\n--- STARTING LIVE MULTI-EXPERT TRACE ---\n".to_string());
    // async for
    while let Some(_) = arb.get_cot_response(query, sys_prompt, /* verbose= */ false).next().await {
        // pass
    }
}
